### ISSUE TITLE
Protocol with generic kwargs-only argument raises reportGeneralTypeIssues

### ISSUE THREAD

**Issue by Finistere** [microsoft/pyright#3379]
**Describe the bug**
Errors when using a protocol with a generic kwargs-only argument. `mypy --strict` (version `0.942`) works on the following example and surprisingly it only fails with kwargs-only arguments. Using `run(self, value: Tct)` works as intended. 

**To Reproduce**
Code to typecheck:

```python
from typing import Generic, Protocol, TypeVar

T = TypeVar('T')
Tct = TypeVar('Tct', contravariant=True)


class Interface(Protocol[Tct]):
    def run(self, *, value: Tct) -> object:
        ...


class Implementation(Generic[Tct]):
    def run(self, value: Tct) -> object:
        return None


def get(_: T) -> Interface[T]:
    return Implementation[T]()

```

Using kwargs-only arguments in `Implementation` also fails:

```python
class Implementation(Generic[Tct]):
    def run(self, *, value: Tct) -> object:
        return None
```

**Expected behavior**
No errors

**Screenshots or Code**
```
scratch_1.py
  scratch_1.py:18:12 - error: Expression of type "Implementation[T@get]" cannot be assigned to return type "Interface[T@get]"
    "Implementation[T@get]" is incompatible with protocol "Interface[T@get]"
      "run" is an incompatible type
        Type "(value: T@get) -> object" cannot be assigned to type "(*, value: Tct@Interface) -> object"
          Keyword parameter "value" of type "Tct@Interface" cannot be assigned to type "T@get"
            Type "Tct@Interface" cannot be assigned to type "T@get" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Command line, version `1.1.239` with Python `3.9.11`.

**Comment by erictraut** [comment]
I agree this should type check without errors. I'll take a deeper look at the cause of the problem.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

The problem affected any protocol that contained a method with a keyword-only argument that was annotated with a class-scoped TypeVar.

**Comment by Finistere** [comment]
Thanks for the fix!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.240, which I just published. It will also be included in the next release of pylance.
